src/inference/include/openvino/runtime/properties.hpp

```diff
@@ -0,0 +1,638 @@
+     * @tparam Properties Should be the pack of `std::pair<std::string, ov::Any>` types
+     * @param device_name device plugin alias
+     * @param configs Optional pack of pairs: (config parameter name, config parameter value)
+     * @return Pair of string key representation and type erased paramter.
+     */
+    template <typename... Properties>
+    inline util::EnableIfAllProperties<std::pair<std::string, Any>, Properties...> operator()(
+        const std::string& device_name,
+        Properties&&... configs) const {
+        return {device_name, AnyMap{std::pair<std::string, Any>{configs...}}};
+    }
+};
+
+/**
+ * @brief Key to pass set of property values to specified device
+ */
+static constexpr Property properties;
+
+/**
+ * @brief Read only property to get a std::string value representing a full device name.
+ */
+static constexpr Key<std::string, PropertyMutability::RO> full_name{"FULL_DEVICE_NAME"};
+
+/**
+ * @brief Read only property which defines the device architecture.
+ */
+static constexpr Key<std::string, PropertyMutability::RO> architecture{"DEVICE_ARCHITECTURE"};
+
+/**
+ * @brief Enum to define possible device types
+ */
+enum class Type {
+    INTEGRATED = 0,  //!<  Device is integrated into host system
+    DISCRETE = 1,    //!<  Device is not integrated into host system
+};
+
+/** @cond INTERNAL */
+inline std::ostream& operator<<(std::ostream& os, const Type& device_type) {
+    switch (device_type) {
+    case Type::DISCRETE:
+        return os << "discrete";
+    case Type::INTEGRATED:
+        return os << "integrated";
+    default:
+        throw ov::Exception{"Unsupported device separability type"};
+    }
+}
+
+inline std::istream& operator>>(std::istream& is, Type& device_type) {
+    std::string str;
+    is >> str;
+    if (str == "discrete") {
+        device_type = Type::DISCRETE;
+    } else if (str == "integrated") {
+        device_type = Type::INTEGRATED;
+    } else {
+        throw ov::Exception{"Unsupported device type: " + str};
+    }
+    return is;
+}
+/** @endcond */
+
+/**
+ * @brief Read only property to get a type of device. See Type enum definition for possible return values
+ */
+static constexpr Key<Type, PropertyMutability::RO> type{"DEVICE_TYPE"};
+
+/**
+ * @brief Read only property which defines Giga OPS per second count (GFLOPS or GIOPS) for a set of precisions supported
+ * by specified device
+ */
+static constexpr Key<std::map<ie::Precision, float>, PropertyMutability::RO> gops{"DEVICE_GOPS"};
+
+/**
+ * @brief  Read only property to get a float of device thermal
+ */
+static constexpr Key<float, PropertyMutability::RO> thermal{"DEVICE_THERMAL"};
+
+}  // namespace device
+
+/**
+ * @brief The key for enabling of dumping the topology with details of layers and details how
+ * this network would be executed on different devices to the disk in GraphViz format.
+ */
+static constexpr Key<bool, PropertyMutability::RW> dump_graph_dot{"HETERO_DUMP_GRAPH_DOT"};
```

Comment:
Is it really needed here? Maybe it should be moved to some hetero-specific header or at least put into `hetero` namespace?

Reply:
I was also thinking about hereo namespace..

Reply:
will be removed